parser/src/powdr.lalrpop

```diff
@@ -619,35 +633,35 @@ GenericReference: NamespacedPolynomialReference = {
     <path:GenericSymbolPath> => NamespacedPolynomialReference{path: path.0, type_args: path.1},
 }
 
-MatchExpression: Box<Expression> = {
-    <start:@L> "match" <scrutinee:BoxedExpression> "{" <arms:MatchArms> "}" <end:@R> => ctx.to_expr_with_source_ref(MatchExpression{scrutinee, arms}, start, end),
+MatchExpression<StructOption>: Box<Expression> = {
+    <start:@L> "match" <scrutinee:BoxedExpression<TermWithoutStruct>> "{" <arms:MatchArms<StructOption>> "}" <end:@R> => ctx.to_expr_with_source_ref(MatchExpression{scrutinee, arms}, start, end),
```

Comment:
Do we actually need to "forward" the StructOption here to MatchArms (and many other places)? Isn't it only relevant for the Term?

Reply:
You are right, several `StructOption` were not necessary. I simplified the grammar by removing all these cases and leaving `Expression<TermWithoutStruct>` on the edge cases (match, if, and some other error-producing rules like Linkdeclaration). All other cases are now Expression<TermWithStruct>.